src/ca.cpp

```diff
@@ -95,9 +95,24 @@ X509Certificate CertificateAuthority::_signCSR(const CertificateSigningRequest &
 
     _signCertificate(newCertificate.get(), _privateKey, signParams);
 
-    //Sanity check: certificate must be verifiable now
+    // Sanity check: certificate must be verifiable now
     try {
         auto cert = X509Certificate{std::move(newCertificate)};
+        // CA's notBefore can't be smaller than issued cert's notBefore i.e. CA can't
+        // issue a certificate older than CA itself
+        if (this->getRootCertificate().getNotBeforeAsn1() > cert.getNotBeforeAsn1()) {
```

Comment:
This project doesn't explictly use this-> to access members and we normally also don't use the public accessors if they just refer to an internal object.